lightning/src/chain/package.rs

```diff
@@ -809,6 +804,20 @@ impl PackageTemplate {
 			height_original,
 		}
 	}
+
+	fn map_output_type_flags(input_solving_data: &PackageSolvingData) -> (PackageMalleability, bool) {
+		let (malleability, aggregable) = match input_solving_data {
+			PackageSolvingData::RevokedOutput(RevokedOutput { is_counterparty_balance_or_non_anchors: None, .. }) => { (PackageMalleability::Malleable, true) },
+			PackageSolvingData::RevokedOutput(RevokedOutput { opt_anchors: Some(..), .. }) => { (PackageMalleability::Malleable, false) },
+			PackageSolvingData::RevokedOutput(RevokedOutput { opt_anchors: None, .. }) => { (PackageMalleability::Malleable, true) },
```

Comment:
According to the compiler this is an unreachable pattern on the second case? Generally I think it’s better to have something more verbose and matching one-to-one the package case rather than enumeration that might subtlety break if there is change in call tree.

Reply:
```
warning: unreachable pattern
   --> lightning/src/chain/package.rs:493:4
    |
493 |             PackageSolvingData::RevokedOutput(RevokedOutput { opt_anchors: Some(..), is_counterparty_balance_or_non_anchors: Some(..), .. }) => { (PackageMalleability::Malleable, false) },
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(unreachable_patterns)]` on by default

warning: `lightning` (lib) generated 5 warnings (run `cargo fix --lib -p lightning` to apply 1 suggestion)
    Finished dev [optimized + debuginfo] target(s) in 0.03s
```

Note I’m on 1.69.0 ?